src/rider/main/kotlin/com/jetbrains/rider/plugins/unreal/actions/PlaySettingsActions.kt

```diff
@@ -174,6 +351,33 @@ class PlayMode : DumbAwareToggleAction() {
 
     override fun update(e: AnActionEvent) {
         super.update(e)
-        e.presentation.isEnabledAndVisible = e.getHost()?.isConnectedToUnrealEditor ?: false
+
+        updatePlayActionPresentation(e)
+    }
+}
+
+class HidePlayButtonsAction : DumbAwareToggleAction() {
+    override fun getActionUpdateThread(): ActionUpdateThread = ActionUpdateThread.BGT
+
+    override fun update(e: AnActionEvent) {
+        super.update(e)
+
+        val host = e.getUnrealHost()
+        if (host == null) {
+            e.presentation.isEnabledAndVisible = false
+        }
+        e.presentation.isEnabledAndVisible = true
```

Comment:
Looks like there's a `return` missing.
But I'd prefer here (and similar places) `isEnabledAndVisible = host != null`